docs/security/selinux/a-beginners-guide-to-selinux-on-centos-7/index.md

```diff
@@ -0,0 +1,227 @@
+
+{{< note >}}
+If SELinux is currently disabled, update your SELinux configuration file with the `SELINUX` directive set to `enabled`, then reboot your system, and SSH back into your Linode. These steps are outlined in the [SELinux States](#selinux-states) section of the guide.
+{{</ note >}}
+
+ - In enforcing mode, SELinux enforces its policies on your system and denies access based on those policies. Use the following command to view SELinux policy modules currently loaded into memory:
+
+        sudo semodule -l
+
+ - Permissive mode does not enforce any of your SELinux policies, instead, it logs any actions that would have been denied to your `/var/log/audit/audit.log` file.
+
+- You can check which mode your system is running by issuing the following command:
+
+        sudo getenforce
+
+- To place SELinux in permissive mode, use the following command:
+
+        sudo setenforce 0
+
+    Permissive mode is useful when configuring your system, because you and your system's components can interact with your files, scripts, and programs without restriction. However, you can use audit logs and system messages to understand what would be restricted in enforcing mode. This will help you better construct the necessary policies for your system's user's and programs.
+
+- Use the `sealert` utility to generate a report from your audit log. The log will include information about what SELinux is preventing and how to allow the action, if desired.
+
+        sudo sealert -a /var/log/audit/audit.log
+
+    Your output will resemble the example, however, it will vary depending on the programs and configurations on your system.
+
+    {{< output >}}
+SELinux is preventing /usr/sbin/httpd from write access on the directory logs.
+
+*****  Plugin httpd_write_content (92.2 confidence) suggests   ***************
+
+If you want to allow httpd to have write access on the logs directory
+Then you need to change the label on 'logs'
+Do
+# semanage fcontext -a -t httpd_sys_rw_content_t 'logs'
+# restorecon -v 'logs'
+    {{</ output >}}
+
+- The example was generated using a [Linode running the Apache webserver](/docs/web-servers/lamp/how-to-install-a-lamp-stack-on-centos-7/#apache) with a virtual hosts configuration.
+
+- To allow `/usr/sbin/httpd` write access to the directory logs, as shown by the output, you would issue the suggested commands, `semanage fcontext -a -t httpd_sys_rw_content_t 'logs'` and `restorecon -v 'logs'`.
+
+## SELinux Context
+
+SELinux marks every single object on a machine with a *context*. Every file, user, and process has a context. The context is broken into three parts: *user*, *role*, and *type*. An SELinux policy controls which users can get which roles. Each specific role places a constraint on what type of files that user can access. When a user logs in to a system, a role is assigned to the user. which can be seen in the `ls -Z` example above: the output `unconfined_u` is a user role.
```

Comment:
Errant period after "a role is assigned to the user." Probably meant to be a comma.